warp10/src/main/java/io/warp10/script/functions/POLYFUNC.java

```diff
@@ -0,0 +1,198 @@
+//
+//   Copyright 2018-2021  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.script.functions;
+
+import io.warp10.continuum.gts.GTSDecoder;
+import io.warp10.continuum.gts.GTSEncoder;
+import io.warp10.continuum.gts.GTSHelper;
+import io.warp10.continuum.gts.GeoTimeSerie;
+import io.warp10.script.NamedWarpScriptFunction;
+import io.warp10.script.StackUtils;
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptLib;
+import io.warp10.script.WarpScriptMapperFunction;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.script.WarpScriptStackFunction;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.commons.math3.analysis.polynomials.PolynomialFunction;
+import org.apache.kafka.common.config.ConfigDef.Type;
+
+/**
+ * Build a function which will compute a polynomial function
+ */
+public class POLYFUNC extends NamedWarpScriptFunction implements WarpScriptStackFunction, WarpScriptMapperFunction {
+
+  private final PolynomialFunction func;
+
+  private final boolean useBucketId;
+
+  public static class Builder extends NamedWarpScriptFunction implements WarpScriptStackFunction {
+
+    public Builder(String name) {
+      super(name);
+    }
+
+    @Override
+    public Object apply(WarpScriptStack stack) throws WarpScriptException {
+      Object top = stack.pop();
+
+      boolean useBucketId = false;
+
+      if (top instanceof Boolean) {
```

Comment:
Well what I had in mind was that this check would rather not done by the builder, but instead by the function that is created by POLYFUNC, so it would be more flexible (the function could then still be applied on other input type than bucketized GTS)